website/source/docs/operating-a-job/configuring-tasks.html.md

```diff
@@ -99,15 +101,76 @@ job "docs" {
 }
 ```
 
+## Environment Variables
+
+Some applications can be configured via environment variables. [The
+Twelve-Factor App](https://12factor.net/config) document suggests configuring
+applications through environment variables. Nomad supports custom environment
+variables in two ways:
+
+* Interpolation in an `env` stanza
+* Templated in the a `template` stanza
+
+### `env` stanza
+
+Each task may have an `env` stanza which specifies environment variables:
+
+```hcl
+task "server" {
+  env {
+    my_key = "my-value"
+  }
+}
+```
+
+The `env` stanza also supports
+[interpolation](/docs/runtime/interpolation.html):
+
+```hcl
+task "server" {
+  env {
+    LISTEN_PORT = "${NOMAD_PORT_http}"
+  }
+}
+```
+
+See the [`env`](/docs/job-specification/env.html.md) docs for details.
+
+
+### Environment Templates
+
+Nomad's [`template`][template] stanza can be used
+to generate environment variables. Environment variables may be templated with
+the contents of files on disk, Consul keys, or secrets from Vault:
+
+```hcl
+template {
+  data = <<EOH
+LOG_LEVEL="{{key "service/geo-api/log-verbosity"}}"
+API_KEY="{{with secret "secret/geo-api-key"}}{{.Data.key}}{{end}}"
+CERT={{ file "path/to/cert.pem" | to JSON }}
```

Comment:
I would also mention that even these env vars can be interpolated. It is super nice but may not be obvious.